### ISSUE TITLE
Using EAGLE will slow down inference

### ISSUE THREAD

**Issue by zkqq** [SafeAILab/EAGLE#73]
Thank you very much for your work on EAGLE; it has been extremely helpful to me.

I have a question: why does downloading _yuhuili/EAGLE-Vicuna-7B-v1.3_ from Hugging Face and using it directly to accelerate _lmsys/vicuna-7b-v1.3_ result in a negative effect? However, using my own trained EAGLE head produces a speedup effect. Could you please tell me where I went wrong?

Below is a screenshot of my operation.

I would greatly appreciate any assistance you can provide in resolving this issue. Thank you very much.

![image](https://github.com/SafeAILab/EAGLE/assets/91409913/0ceb7ec3-8b18-4516-ace3-3df156b8f3bf)
![image](https://github.com/SafeAILab/EAGLE/assets/91409913/5a6a974a-cf5b-4bf1-a77a-39de5df3f5cc)
![image](https://github.com/SafeAILab/EAGLE/assets/91409913/1d7da122-66e5-4b5f-971a-56447b78c9d8)
![image](https://github.com/SafeAILab/EAGLE/assets/91409913/b5e79e77-f036-4177-a603-cfe92fe0b47e)
![image](https://github.com/SafeAILab/EAGLE/assets/91409913/cfb9cce2-4e1c-4018-87be-c4147deaf77e)

**Comment by cdliang11** [comment]
Maybe try temperature=0.

**Comment by zkqq** [comment]
> Maybe try temperature=0.

Thank you very much for your valuable advice. However, I obtained the same result regardless of the temperature.

![image](https://github.com/SafeAILab/EAGLE/assets/91409913/4e478f88-f572-4b3b-a3ea-75b9dde54cd9)
![image](https://github.com/SafeAILab/EAGLE/assets/91409913/149de1dc-fa0d-477b-99e3-8cb3d104cee2)

**Comment by Liyuhui-12** [comment]
@zkqq  The correct drafts should be highlighted in yellow. I noticed that your images contain almost no yellow words. You might not have correctly matched the draft model with the base model or not used the template correctly (the --model-type parameter).

**Comment by zkqq** [comment]
> yuhuili/EAGLE-Vicuna-7B-v1.3

Thank you very much for your reply. You are correct; the issue likely stems from the mismatch between the EAGLE head and the origin model. However, I believe I have configured all necessary parameters, including the model type.

I trained an EAGLE head, ran webui.py and the evaluation, and observed a good acceleration effect. However, when switching back to the EAGLE head from _yuhuili/EAGLE-Vicuna-7B-v1.3_, there are negative impacts. Both _config.json_ are identical, with the only difference being the _pytorch_model.bin_ file.
